python/extensions/pip.bzl

```diff
@@ -428,6 +429,18 @@ cannot have a child module that uses the same `hub_name`.
     }
     return attrs
 
+def _extension_extra_args():
+    args = {}
+
+    if bazel_features.external_deps.module_extension_has_os_arch_dependent:
+        args = args | {
+            "arch_dependent": True,
+            "os_dependent": True,
+        }
+        fail("This does not work right now")
```

Comment:
@fmeum, could you help me understand why the `bazel_features` detection does not work? I added a `fail` statement here, but
```
$ USE_BAZEL_VERSION=last_green bazel test --lockfile_mode=update ...
```

is completing successfully and the lock file does not have the new structure.

Reply:
Where did you run this? If I run it in `examples/bzlmod`, I get the expected failure. When I comment out the `fail`, I do get the expected lockfile structure.

Reply:
It seems that I needed to do `bazel clean` in order to get it to work. I have seen similar behaviour in the past where the updated module code would not necessarily work as intended in the `rules_python` code when the external repository content changes.

Reply:
Could you try to provide instructions to reproduce this behavior? This would be a very concerning bug that we would very much like to fix before Bazel 7.

Reply:
I can repro it on my local machine with:
```console
git checkout main
git log -n 3 --oneline
git checkout f2a4dd5
cd examples/bzlmod
USE_BAZEL_VERSION=latest bazel test --lockfile_mode=update ...
# press Ctrl-C to abbort requirements_test
git log -n 3 --oneline main
git checkout cdb5902
USE_BAZEL_VERSION=last_green bazel test --lockfile_mode=update ...
# press Ctrl-C to abbort requirements_test
```

Reply:
And then later I can reproduce it with just:
```console
git checkout f2a4dd5
USE_BAZEL_VERSION=last_green bazel test --lockfile_mode=update ...
# press Ctrl-C to abort requirements_test
git checkout main
USE_BAZEL_VERSION=last_green bazel test --lockfile_mode=update ...
```